testing/opae-cxx/test_object_cxx_core.cpp

```diff
@@ -47,21 +47,36 @@ class sysobject_cxx_p : public ::testing::TestWithParam<std::string> {
     invalid_device_ = test_device::unknown();
 
     ASSERT_EQ(fpgaInitialize(NULL), FPGA_OK);
-    tokens_ = token::enumerate({properties::get(FPGA_ACCELERATOR)});
+
+    properties::ptr_t props = properties::get(FPGA_ACCELERATOR);
+    props->device_id = platform_.devices[0].device_id;
+
+    tokens_ = token::enumerate({props});
     ASSERT_GT(tokens_.size(), 0);
     handle_ = handle::open(tokens_[0], 0);
     ASSERT_NE(handle_.get(), nullptr);
+
+    properties::ptr_t props_dev = properties::get(FPGA_DEVICE);
+    props_dev->device_id = platform_.devices[0].device_id;
+
+    tokens_dev_ = token::enumerate({props_dev});
+    ASSERT_GT(tokens_dev_.size(), 0);
+    handle_dev_ = handle::open(tokens_dev_[0], 0);
+    ASSERT_NE(handle_dev_.get(), nullptr);
   }
 
   virtual void TearDown() override {
     system_->finalize();
```

Comment:
For similar reasons as the C API, shouldn't system_finalize(); appear after objects are destroyed?